src/main/java/net/minecraftforge/event/world/BlockEvent.java

```diff
@@ -166,25 +168,53 @@ public void setExpToDrop(int exp)
     }
 
     /**
-     * Called when a block is placed by a player.
+     * Called when a block is placed.
      *
      * If a Block Place event is cancelled, the block will not be placed.
      */
     @Cancelable
-    public static class PlaceEvent extends BlockEvent
+    public static class BasePlaceEvent extends BlockEvent
     {
-        private final EntityPlayer player;
+        private final Entity entity;
         private final BlockSnapshot blockSnapshot;
         private final IBlockState placedBlock;
         private final IBlockState placedAgainst;
+
+        public BasePlaceEvent(@Nonnull BlockSnapshot blockSnapshot, @Nonnull IBlockState placedAgainst, @Nullable Entity entity)
+        {
+            super(blockSnapshot.getWorld(), blockSnapshot.getPos(), !(entity instanceof EntityPlayer) ? blockSnapshot.getReplacedBlock() : blockSnapshot.getCurrentBlock());
+            this.entity = entity;
+            this.blockSnapshot = blockSnapshot;
+            this.placedBlock = !(entity instanceof EntityPlayer) ? blockSnapshot.getReplacedBlock() : blockSnapshot.getCurrentBlock();
+            this.placedAgainst = placedAgainst;
+
+            if (DEBUG)
+            {
+                System.out.printf("Created BasePlaceEvent - [PlacedBlock: %s ][PlacedAgainst: %s ][Entity: %s ]\n", getPlacedBlock(), placedAgainst, entity);
+            }
+        }
+
+        @Nullable
+        public Entity getEntity() { return entity; }
+        public BlockSnapshot getBlockSnapshot() { return blockSnapshot; }
+        public IBlockState getPlacedBlock() { return placedBlock; }
+        public IBlockState getPlacedAgainst() { return placedAgainst; }
+    }
+
+    /**
+     * Called when a block is placed by a player.
+     *
+     * If a Block Place event is cancelled, the block will not be placed.
+     */
+    @Cancelable
+    public static class PlaceEvent extends BasePlaceEvent
```

Comment:
Deprecate this and mark for removal in 1.13, no need to have two events for almost the same purpose other than backwards compat. In 1.13 this can just be an `@Nullable getPlayer()` or `Optional<EntityPlayer>` return or something along those lines.
```suggestion
    @Deprecated // Remove in 1.13
    public static class PlaceEvent extends BasePlaceEvent
```
Same applies to the multi version.